CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword
+
+```csharp
+// Bad
+String.IsNullOrEmpty(name);
+
+// Good
+string.IsNullOrEmpty(name);
+```
+
+### Use readonly fields when possible
+
+### Use string interpolation
+
+```csharp
+// Bad
+var message = string.Format("My name is {0} {1}.", person.FirstName, person.LastName);
+
+// Good
+var message = $"My name is {person.FirstName} {person.LastName}.");
+```
+
+## Layout
+
+### Use *four spaces* per indentation level
+
+```csharp
+// Bad
+public void SomeMethod()
+{
+  DoSomethingFirst();
+  DoSomerthingLater();
+}
+
+// Good
+public void SomeMethod()
+{
+    DoSomethingFirst();
+    DoSomerthingLater();
+}
+```
+
+### Remove unused or redudant namespaces
+
+### Use single line and lambda getters
+
+```csharp
+// Good
+public string Name { get; private set; }
+
+// Good
+public string UppercaseName => Name.toUpperCase(); 
+```
+
+### Don't use braces for just one line

```

Comment:
Well, a quick googling did not me examples about code merge resolution, only more mentions about it hehe.

But I found other reasons.

* ASP.NET team use them, for example [ActionDescriptorExtensions](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Abstractions/Abstractions/ActionDescriptorExtensions.cs#L22), [FilterContext](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/FilterContext.cs#L24), [ModelBindingResult](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelBindingResult.cs#L59), [MvcServiceCollectionExtensions](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc/MvcServiceCollectionExtensions.cs#L27), the are using early returns and in my opinion it has not readibility issues. Maybe the code is more extense vertically, but I think that it is not a problem.

* It is [less error prone](https://softwareengineering.stackexchange.com/questions/16528/single-statement-if-block-braces-or-no/16530#answer-16530)

    ```csharp
    if(condition) 
    //    statement;
    otherStatement;
    ```

    ```csharp
    if(condition) 
        statement;
        otherStatement;
    ```